src/shell_pthreads.js

```diff
@@ -4,10 +4,10 @@
 // 3) We could be an application pthread running in a worker. (ENVIRONMENT_IS_WORKER == true and ENVIRONMENT_IS_PTHREAD == true)
 #if USE_PTHREADS
 
-if (typeof ENVIRONMENT_IS_PTHREAD === 'undefined') {
+var ENVIRONMENT_IS_PTHREAD = typeof importScripts === 'function';
```

Comment:
This seems to contradict the commend above.   Not all workers are pthreads, but this change looks like it would set ENVIRONMENT_IS_PTHREAD for all workers.

Reply:
Crap you're right. I'll have to look at another way of determining if the script is being called from a thread.

Reply:
I've changed what I'm doing here, instead of trying to determine if we're in a thread from a script, I'm attaching ENVIRONMENT_IS_PTHREAD to the module that gets passed from the worker.
I'm pretty sure that's the right way to do this?